app/src/main/java/com/magikarp/android/ui/posts/CreatePostFragment.java

```diff
@@ -0,0 +1,105 @@
+package com.magikarp.android.ui.posts;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageView;
+
+import com.magikarp.android.R;
+
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+
+/**
+ * A simple {@link Fragment} subclass.
+ * Use the {@link CreatePostFragment#newInstance} factory method to
+ * create an instance of this fragment.
+ */
+public class CreatePostFragment extends Fragment {
```

Comment:
Recommend a couple of additions:

- This class can probably cover both "new" and "edit" functions based on whether the fragments receives arguments or not.  The logic and UI should be basically the same.  Based on app flow diagram, this fragment would follow the screen where a user is viewing all their posts and either selects an existing post or creates a new post.  The calling fragment should provide the arguments for populating UI for editing existing post.
- Add a top toolbar menu with the option to add/edit geotag.  Can find icons at https://material.io/resources/icons.  Recommend toggling between two icons (ex. "add_location", "edit_location" depending whether there is a geotag already stored.
- Recommend either making "Post" button part of the top menu or aliasing functionality on popup keyboard so user not required to scroll to bottom of UI to post.
- Look at ViewModel and/or saved state for persisting UI across fragment destruction (device rotation, app switching, etc.).
- Maybe alternate layout for landscape orientation (ex. picture on left, text on right, ,etc.) to prevent UI from getting too long.